staging/src/k8s.io/apiextensions-apiserver/examples/client-go/pkg/client/informers/externalversions/factory.go

```diff
@@ -112,39 +107,38 @@ func NewSharedInformerFactoryWithOptions(client versioned.Interface, defaultResy
 	return factory
 }
 
-func (f *sharedInformerFactory) Start(stopCh <-chan struct{}) {
+func (f *sharedInformerFactory) start(stopCh <-chan struct{}) *sync.WaitGroup {
+	wg := &sync.WaitGroup{}
 	f.lock.Lock()
 	defer f.lock.Unlock()
 
-	if f.shuttingDown {
-		return
-	}
-
 	for informerType, informer := range f.informers {
 		if !f.startedInformers[informerType] {
-			f.wg.Add(1)
-			// We need a new variable in each loop iteration,
-			// otherwise the goroutine would use the loop variable
-			// and that keeps changing.
-			informer := informer
+			wg.Add(1)
+			informerCopy := informer
 			go func() {
-				defer f.wg.Done()
-				informer.Run(stopCh)
+				defer wg.Done()
+				informerCopy.Run(stopCh)
 			}()
 			f.startedInformers[informerType] = true
 		}
 	}
+
+	return wg
 }
 
-func (f *sharedInformerFactory) Shutdown() {
```

Comment:
This made it out in a release, so we can't just delete it, we have to keep it working. :(

Reply:
:(

make sense to deprecate it?